src/commands/codepush/lib/update-contents-tasks/zip.ts

```diff
@@ -2,6 +2,7 @@ import * as fs from "fs";
 import * as pfs from "../../../../util/misc/promisfied-fs";
 import * as path from "path";
 import * as JsZip from "jszip";
+var yazl = require("yazl");
```

Comment:
No typings? 

If you have to use require, use `const` for the variable, not `var`.

And what is the compelling value of yazl over jszip that you need to bring in a second package with repeated functionality?

Reply:
There is no original typings, so I've added custom typings from the old code-push-cli https://github.com/Microsoft/code-push/blob/master/definitions/yazl.d.ts

Regarding the repeated functionality - this hacky solution is needed to consistent hashes calculation both on codepush clients SDKs and on the old codepush service. Old codepush service uses yazl lib to work with zip archives and for some reason the hashes are calculated wrong for the zip packages produced with jszip lib, but works correctly with yazl.

So I'm absolutely agreed that the yazl contains repeated functionality but unfortunately we need it until we not figured out what the difference between yazl and jszip in terms of hashes calculation. Another words it's a temporary solution that hopefully will be removed soon.

Reply:
Fair enough, that's a reasonable explanation. Thanks!

Although - how do we know that yazl wasn't doing it wrong rather than jszip? 😄 As long as it works, I guess it's ok.